### PR TITLE
Feat/health endpoint from v20

### PR DESCRIPTION
**Pull request by annasambrook**

### CODE REVIEW

**Review by dagacha — APPROVED** [review]
LGTM - tested

File: packages/valory/skills/decision_maker_abci/states/decision_receive.py [hunk 1]
```diff
@@ -59,6 +59,12 @@ def end_block(self) -> Optional[Tuple[SynchronizedData, Enum]]:
             return None
 
         synced_data, event = cast(Tuple[SynchronizedData, Enum], res)
+        decision_receive_timestamp = self.most_voted_payload_values[-1]
+
+        synced_data = cast(
+            SynchronizedData,
+            synced_data.update(decision_receive_timestamp=decision_receive_timestamp),
+        )
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/states/decision_receive.py:67]: Just checking: do we want to update the timestamp in all cases? Even if mech response failed?

File: packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py [hunk 2]
```diff
@@ -516,6 +516,7 @@ def async_act(self) -> Generator:
                 prediction_response.confidence if prediction_response else None,
                 bet_amount,
                 next_mock_data_row,
+                decision_received_timestamp=self.synced_timestamp,
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py:519]: :warning: Here you disable the NONE event!! It cannot be triggered anymore!

File: packages/valory/skills/decision_maker_abci/handlers.py [hunk 3]
```diff
@@ -337,3 +347,19 @@ def _send_not_found_response(
...
+            > self.context.params.agent_balance_threshold
+        )
+
+    def _check_is_receiving_mech_responses(self) -> bool:
+        """Check the agent is making on chain transactions."""
+        # Checks the most recent decision receive timestamp, which can only be returned after making a mech call
+        # (an on chain transaction)
+        return (
+            self.synchronized_data.decision_receive_timestamp
+            < 2 * self.context.params.reset_pause_duration
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/handlers.py:364]: Why compared with the reset and pause duration? Here we are saying:

if the mech response takes longer than twice the **duration** of the reset and pause round then we are not receiving mech responses. IMO we should introduce a separate param and set it to the expected mech time response.

File: packages/valory/skills/decision_maker_abci/behaviours/bet_placement.py [hunk 4]
```diff
@@ -228,8 +231,12 @@ def async_act(self) -> Generator:
...
 
-            payload = MultisigTxPayload(
-                agent, tx_submitter, betting_tx_hex, mocking_mode
+            payload = BetPlacementPayload(
+                agent,
+                tx_submitter,
+                betting_tx_hex,
+                mocking_mode,
+                wallet_balance=self.wallet_balance,
             )
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/behaviours/bet_placement.py:240]: :warning: Here you disable the NONE event!! It cannot be triggered anymore!
Moreover, you are returning two different kinds of payloads in this behaviour!

File: packages/valory/skills/check_stop_trading_abci/rounds.py [hunk 5]
```diff
@@ -73,6 +77,20 @@ class CheckStopTradingRound(VotingRound):
...
 
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
+        """Process the end of the block."""
+        res = super().end_block()
+
+        if res is None:
+            return None
+
+        sync_data, event = res
+
```
Comment (Adamantios) [inline packages/valory/skills/check_stop_trading_abci/rounds.py:88]:
```suggestion

```

File: packages/valory/agents/trader/aea-config.yaml [hunk 6]
```diff
@@ -252,8 +252,6 @@ models:
       collateral_balance: ${int:10000000000000000000}
       mech_cost: ${int:10000000000000000}
       pool_fee: ${int:20000000000000000}
-      outcome_token_amounts: ${list:[11000000000000000000,9000000000000000000]}
-      outcome_token_marginal_prices: ${list:[0.4,0.6]}
```
Comment (Adamantios) [inline packages/valory/agents/trader/aea-config.yaml:256]: Did this leak to this PR somehow?

File: packages/valory/services/trader/service.yaml [hunk 7]
```diff
@@ -155,8 +155,6 @@ type: skill
         collateral_balance: ${BENCHMARKING_MODE_COLLATERAL_BALANCE:int:10000000000000000000}
         mech_cost: ${BENCHMARKING_MODE_MECH_COST:int:10000000000000000}
         pool_fee: ${BENCHMARKING_MODE_POOL_FEE:int:20000000000000000}
-        outcome_token_amounts: ${OUTCOME_TOKEN_AMOUNTS:list:[11000000000000000000,9000000000000000000]}
-        outcome_token_marginal_prices: ${OUTCOME_TOKEN_MARGINAL_PRICES:list:[0.55,0.45]}
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:159]: Did this leak to this PR somehow?

**Review by Adamantios — CHANGES_REQUESTED** [review]

File: packages/valory/skills/check_stop_trading_abci/rounds.py (hunk 5 above)
Comment (annasambrook) [reply packages/valory/skills/check_stop_trading_abci/rounds.py:88]: I don't understand why this should be removed?

File: packages/valory/skills/decision_maker_abci/states/decision_receive.py (hunk 1 above)
Comment (annasambrook) [reply packages/valory/skills/decision_maker_abci/states/decision_receive.py]: This is now only updated in the case of a DONE event

File: packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py (hunk 2 above)
Comment (annasambrook) [reply packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py]: Should be fixed now.

File: packages/valory/skills/decision_maker_abci/handlers.py (hunk 3 above)
Comment (annasambrook) [reply packages/valory/skills/decision_maker_abci/handlers.py]: Should be done now.

File: packages/valory/skills/decision_maker_abci/behaviours/bet_placement.py (hunk 4 above)
Comment (annasambrook) [reply packages/valory/skills/decision_maker_abci/behaviours/bet_placement.py]: NONE event issue should be fixed. Also updated the type of payload.

File: packages/valory/agents/trader/aea-config.yaml (hunk 6 above)
Comment (annasambrook) [reply packages/valory/agents/trader/aea-config.yaml]: No idea how this got removed, but its now back.

File: packages/valory/services/trader/service.yaml (hunk 7 above)
Comment (annasambrook) [reply packages/valory/services/trader/service.yaml]: No idea how this got removed, but its now back.

File: packages/valory/skills/check_stop_trading_abci/rounds.py (hunk 5 above)
Comment (Adamantios) [reply packages/valory/skills/check_stop_trading_abci/rounds.py]: Is it used in any way?

File: packages/valory/services/trader/service.yaml [hunk 8]
```diff
@@ -145,6 +145,7 @@ type: skill
         rebet_chance: ${REBET_CHANCE:float:0.6}
         mech_interaction_sleep_time: ${MECH_INTERACTION_SLEEP_TIME:int:10}
         policy_store_update_offset: ${POLICY_STORE_UPDATE_OFFSET:int:259200}
+        expected_mech_response_time: ${EXPECTED_MECH_RESPONSE_TIME:int:300}
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:148]: This is not in the agent's configuration.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**